lib/auth/oidc.go

```diff
@@ -78,36 +79,23 @@ func (a *Server) createOIDCClient(conn types.OIDCConnector) (*oidc.Client, error
 		return nil, trace.Wrap(err)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaults.WebHeadersTimeout)
-	defer cancel()
-
+	doneSyncing := make(chan struct{})
 	go func() {
-		defer cancel()
+		defer close(doneSyncing)
 		client.SyncProviderConfig(conn.GetIssuerURL())
 	}()
 
 	select {
-	case <-ctx.Done():
+	case <-doneSyncing:
+	case <-time.After(defaults.WebHeadersTimeout):
+		return nil, trace.ConnectionProblem(nil,
+			"timed out syncing oidc connector %v, "+
+				"ensure URL %q is valid and accessible and check configuration",
+			conn.GetName(), conn.GetIssuerURL())
 	case <-a.closeCtx.Done():
 		return nil, trace.ConnectionProblem(nil, "auth server is shutting down")
 	}
 
-	// Canceled is expected in case if sync provider config finishes faster
```

Comment:
Note to reviewers: technically this change isn't strictly required, but I thought it was odd to use a context in this way. Additionally, `ctx.Done()` was used for both the happy path and the timeout case, which I found difficult to reason about.

This new approach has 3 distinct cases, one for each possible scenario:
- everything worked
- we timed out
- the auth server is shutting down

Reply:
Seems reasonable. I assume we are also covered by tests, so no objections on my part.